tests/unit/test_expressions.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+from aiodynamo.expressions import F, Parameters
+
+
+@pytest.mark.parametrize(
+    "pe,expression,names",
+    [
+        (F("foo") & F("bar"), "#n0,#n1", {"#n0": "foo", "#n1": "bar"}),
+        (F("foo", 0, "bar") & F("bar"), "#n0[0].#n1,#n1", {"#n0": "foo", "#n1": "bar"}),
+        (
+            F("foo", "12", "bar") & F("bar"),
+            "#n0.#n1.#n2,#n2",
+            {"#n0": "foo", "#n1": "12", "#n2": "bar"},
```

Comment:
This is kinda hard to read... I guess it's like `a+b=c`? or more like `a→b,c`?

Reply:
which part is hard to read? I don't know what a, b and c are.

Reply:
Eh to me (someone who didn't write the code) this looks roughly like this:
```
@pytest.mark.parametrize(
    "pe,expression,names",
    [
        %%%%%%%%%% ####### ###%###%###%##%###%
        %%%%%%%%%%%% ######## ####%####%####%#####%
            %%%%%%%%%%%%%
            ###########
            #####%####%#####%####%####%
```

I do blame amazon for the crazy syntax with all the oddball non-alpha characters...
TBH I don't know if how this could be improved.

One thing that was hard was to figure out which of the pytest parameters (pr, exp, nam) were inputs and which outputs.